app/services/credit_notes/estimate_service.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module CreditNotes
+  class EstimateService < BaseService
+    def initialize(invoice:, items:)
+      @invoice = invoice
+      @items = items
+
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: 'invoice') unless invoice
+      return result.forbidden_failure! unless License.premium?
+      return result.not_allowed_failure!(code: 'invalid_type_or_status') unless valid_type_or_status?
+
+      credit_note = CreditNote.new(
```

Comment:
It seems to me that `credit_note` could be an instance variable in this service based on context and that we can avoid passing it in many methods as a parameter. wdyt?